packages/demo/src/app.ts

```diff
@@ -127,17 +164,50 @@ smartFilterLoader.manifests.forEach((manifest) => {
 // Set up SmartFilter <select> handler
 smartFilterSelect.addEventListener("change", () => {
     localStorage.setItem(LocalStorageSmartFilterName, smartFilterSelect.value);
-    smartFilterLoader.loadFromManifest(smartFilterSelect.value, optimize);
+    try {
+        smartFilterLoader.loadFromManifest(smartFilterSelect.value);
+    } catch (e) {
+        smartFilterLoader.loadFromManifest(smartFilterLoader.defaultSmartFilterName);
+    }
 });
 
 // Set up editor button
 editActionLink.onclick = async () => {
-    if (currentSmartFilter) {
+    if (currentSmartFilterState) {
         const module = await import(/* webpackChunkName: "smartFilterEditor" */ "./helpers/launchEditor");
-        module.launchEditor(currentSmartFilter, engine, renderer, smartFilterLoader);
+        module.launchEditor(
+            editOptimized && currentSmartFilterState.optimizedSmartFilter
+                ? currentSmartFilterState.optimizedSmartFilter
+                : currentSmartFilterState.smartFilter,
+            engine,
+            renderer,
+            smartFilterLoader
+        );
     }
 };
 
+// Set up the optimize checkbox
+optimizeCheckbox.checked = optimize;
+optimizeCheckbox.onchange = () => {
```

Comment:
Good catch - but I think I'll fix it differently - instead of setting editOptimized to false and having to remember if it should be set to true again when optimize becomes true again, I'll just check optimized as well when deciding how to launch the editor.  Thanks!